apps/aehttp/test/aehttp_integration_SUITE.erl

```diff
@@ -3326,32 +3326,51 @@ naming_system_broken_txs(_Config) ->
 %% Channels
 %%
 assert_balance(Pubkey, ExpectedBalance) ->
-    Address = aeser_api_encoder:encode(account_pubkey, Pubkey),
-    {ok, 200, #{<<"balance">> := ExpectedBalance}} =
-        get_accounts_by_pubkey_sut(Address).
+    assert_balance(Pubkey, ExpectedBalance, equal).
+
+assert_balance_at_most(Pubkey, ExpectedBalance) ->
+    assert_balance(Pubkey, ExpectedBalance, equal_or_less).
+
+assert_balance_at_least(Pubkey, MaxExpectedBalance) ->
+    assert_balance(Pubkey, MaxExpectedBalance, equal_or_greater).
 
-assert_balance_at_least(Pubkey, MinExpectedBalance) ->
+assert_balance(Pubkey, ExpectedBalance, Action) ->
     Address = aeser_api_encoder:encode(account_pubkey, Pubkey),
-    {ok, 200, #{<<"balance">> := Balance}} =
+    {ok, 200, #{<<"balance">> := ActualBalance}} =
         get_accounts_by_pubkey_sut(Address),
-    true = MinExpectedBalance =< Balance.
+    Res =
+        case Action of
+            equal -> ExpectedBalance =:= ActualBalance;
+            equal_or_greater -> ActualBalance >= ExpectedBalance;
+            equal_or_less -> ActualBalance =< ExpectedBalance
+        end,
+    {true, _, _, _} = {Res, ActualBalance, Action, ExpectedBalance}.
 
 assert_trees_balance(Trees, Pubkey, ExpectedBalance) ->
     AccTrees = aec_trees:accounts(Trees),
     {value, Account} = aec_accounts_trees:lookup(Pubkey, AccTrees),
     ExpectedBalance = aec_accounts:balance(Account).
 
-channel_sign_tx(ConnPid, Privkey, Method, Config) ->
-    {ok, Tag, #{<<"tx">> := EncCreateTx,
+channel_sign_tx(Pubkey, ConnPid, Privkey, Method, Config) ->
+    {ok, Tag, #{<<"tx">> := EncTx,
                 <<"updates">> := Updates}} = wait_for_channel_event(ConnPid, sign, Config),
     Method = <<"channels.", (bin(Tag))/binary>>,
-    {ok, CreateBinTx} = aeser_api_encoder:safe_decode(transaction, EncCreateTx),
-    Tx = aetx:deserialize_from_binary(CreateBinTx),
-    SignedCreateTx = aec_test_utils:sign_tx(Tx, Privkey),
-    EncSignedCreateTx = aeser_api_encoder:encode(transaction,
-                                  aetx_sign:serialize_to_binary(SignedCreateTx)),
-    ws_call_async_method(ConnPid, Method, #{tx => EncSignedCreateTx}, Config),
-    #{tx => Tx, signed_tx => SignedCreateTx, updates => Updates}.
+    {ok, TxBin} = aeser_api_encoder:safe_decode(transaction, EncTx),
+    Tx = aetx:deserialize_from_binary(TxBin),
+    SignedTx = 
+        case account_type(Pubkey) of
```

Comment:
I see this is changing behaviour of the test depending on the behaviour of the system under test - chain. I see that when test makes account generic it mines with `strictly_follow_top`. Seems good enough. Ok.